lib/common.bash

```diff
@@ -532,3 +532,354 @@ check_dockerfiles_images()
+
+  # Retrieve pod name
+  local pod_name=$(esudo kubectl get pod -o wide | grep ${partial_pod_name} | awk '{print $1;}' || true)
+
+  # Delete by yaml
+  esudo kubectl delete -f "${yaml}" 2>/dev/null || true
+  
+  # Verify pod deleted
+  [ -z "${pod_name}" ] || (k8s_wait_for_pod_delete_state "${pod_name}" || true)
+}
+
+# Retrieve pod name and log kubernetes environment information: 
+# nodes, services, deployments, pods
+k8s_print_info() {
+  local partial_pod_name="${1}"
+  
+  echo "-------------------------------------------------------------------------------"
+  esudo kubectl get nodes -o wide
+  echo "-------------------------------------------------------------------------------"
+  esudo kubectl get services -o wide
+  echo "-------------------------------------------------------------------------------"
+  esudo kubectl get deployments -o wide
+  echo "-------------------------------------------------------------------------------"
+  esudo kubectl get pods -o wide
+  echo "-------------------------------------------------------------------------------"
+  local pod_name=$(esudo kubectl get pod -o wide | grep "${partial_pod_name}" | awk '{print $1;}')
+  esudo kubectl describe pod "${pod_name}"
+  echo "-------------------------------------------------------------------------------"
+}
+
+
+###############################################################################
+
+# kata
+
+# Find sandbox id using container ID
+kata_get_sandbox_id() {
+  local container_id="${1}"
+  local sandbox_dir="/run/kata-containers/shared/sandboxes"
+
+  # Find container directory inside sandbox directory
+  local container_dir=$(esudo find "${sandbox_dir}" -name "${container_id}" | head -1)
+
+  # Ensure directory path pattern is correct
+  [[ ${container_dir} =~ ^${sandbox_dir}/.*/.*/${container_id} ]] \
+    || (>&2 echo "Incorrect container folder path: ${container_dir}"; return 1)
+
+  # Two levels up, and trim the sandbox dir off the front
+  local sandbox_id=$(dirname $(dirname "${container_dir}") | sed "s|${sandbox_dir}/||g")
+
+  echo "${sandbox_id}"
+}
+
+# Get guest kernel append from qemu command line
+kata_get_guest_kernel_append() {
+  local pod_name="${1}"
+  local duration=$((SECONDS+20))
+  local kernel_append
+
+  # Attempt to get qemu command line from qemu process
+  while [ $SECONDS -lt $duration ]; do
+    container_id=$(k8s_get_container_id "${pod_name}")
+    sandbox_id=$(kata_get_sandbox_id "${container_id}")
+    qemu_process=$(ps aux | grep qemu | grep ${sandbox_id} | grep append || true)
+    if [ -n "${qemu_process}" ]; then
+      kernel_append=$(echo ${qemu_process} \
+        | sed "s|.*-append \(.*$\)|\1|g" \
+        | sed "s| -.*$||")
+      break
+    fi
+    sleep 1
+  done
+
+  [ -n "${kernel_append}" ] \
+    || (>&2 echo "Could not retrieve guest kernel append parameters"; return 1)
+
+  echo "${kernel_append}"
+}
+
+
+###############################################################################
+
+# simple-kbs
```

Comment:
I suggest to move the `simple-kbs` functions to `integration/confidential/lib.sh`.

Reply:
Moved.